R/assertions.R

```diff
@@ -132,30 +133,41 @@ assert_data_frame <- function(arg,
 assert_character_scalar <- function(arg,
                                     values = NULL,
                                     case_sensitive = TRUE,
-                                    optional = FALSE) {
+                                    optional = FALSE,
+                                    arg_name = rlang::caller_arg(arg),
+                                    message = NULL,
+                                    class = "assert_character_scalar",
+                                    call = parent.frame()) {
   assert_character_vector(values, optional = TRUE)
   assert_logical_scalar(optional)
 
   if (optional && is.null(arg)) {
     return(invisible(arg))
   }
 
+  message <-
+    message %||%
+    ifelse(
+      is.null(values),
+      "Argument {.arg {arg_name}} must be a scalar of class {.cls character},
+       but is {.obj_type_friendly {arg}}.",
+      "Argument {.arg {arg_name}} must be equal to one of {.val {values}}."
+    )
```

Comment:
I agree we should absolutely talk about the message argument, thanks @bundfussr !

If we were designing these checks from scratch, I would have advocated for simpler checks rather than compound checks. But now that we have compound checks, I am in favor of compound default messages still with the ability to modify the message via an argument. As more admiral-flavored packages are built, they may need to change the messages in certain situations and the message argument gives them that flexibility.

In the example above, I didn't add the class to the message because I thought that by listing the values that are accepted it conveys the information the user needs. If we feel we that we should list the values and the class of these values, that can be done. But I personally don't think that is needed to message to the users how they should update their code.

Reply:
I updated the message to re-iterate that the value must be scalar character.

```r
ifelse(
      is.null(values),
      "Argument {.arg {arg_name}} must be a scalar of class {.cls character},
       but is {.obj_type_friendly {arg}}.",
      "Argument {.arg {arg_name}} must be a scalar of class {.cls character} and
       equal to one of {.val {values}}."
    )
```